api/files/views.py

```diff
@@ -35,7 +35,235 @@ def get_file(self, check_permissions=True):
 
 
 class FileDetail(generics.RetrieveUpdateAPIView, FileMixin):
-    """Details about a specific file.
+    """Details about files and folders. *Writeable*.
+
+    Welcome to the Files API.  Brace yourself, things are about to get *weird*.
+
+    The Files API is the one place in the OSF API where we break hard from the JSON-API spec.  This is because most of
+    the behind-the-scenes moving, uploading, deleting, etc. of files and folders is actually be handled for us by a
+    nifty piece of software called [WaterButler](https://github.com/CenterForOpenScience/waterbutler).  WaterButler lets
+    us interact with files stored on different cloud storage platforms through a consistent API.  However, it uses
```

Comment:
This may be covered later on, but there will be some services that work differently than others in a way that we will need to note. For example, dataverse has this concept of draft and published versions of projects, and switching between them is, I think, a query parameter or something. That might not work as something we provide links for, so we may have to document that. But we should do that as a separate PR; I don't want to hold this up for dataverse.